From 230d9c127d45af11739214e78488a93255b832bb Mon Sep 17 00:00:00 2001
From: Gyorgy Sarvari <skandigraun@gmail.com>
Date: Tue, 16 Sep 2025 16:28:34 +0200
Subject: [PATCH] ignore lockf() return value

gcc15 returns an error otherwise:

| ../sources/libpisp-1.2.1/src/helpers/media_device.cpp: In member function 'std::map<std::__cxx11::basic_string<char>, libpisp::helpers::DeviceFd>::iterator libpisp::helpers::MediaDevice::unlock(const std::string&)':
| ../sources/libpisp-1.2.1/src/helpers/media_device.cpp:251:14: error: ignoring return value of 'int lockf(int, int, __off64_t)' declared with attribute 'warn_unused_result' [-Werror=unused-result]
|   251 |         lockf(it->second.Get(), F_ULOCK, 0);
|       |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Upstream-Status: Submitted [https://github.com/raspberrypi/libpisp/pull/53]

Signed-off-by: Gyorgy Sarvari <skandigraun@gmail.com>
---
 src/helpers/media_device.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/helpers/media_device.cpp b/src/helpers/media_device.cpp
index 0dd64d6..121ce79 100644
--- a/src/helpers/media_device.cpp
+++ b/src/helpers/media_device.cpp
@@ -248,6 +248,6 @@ std::map<std::string, DeviceFd>::iterator MediaDevice::unlock(const std::string
 	if (it == lock_map_.end())
 		return lock_map_.end();
 
-	lockf(it->second.Get(), F_ULOCK, 0);
+	std::ignore = lockf(it->second.Get(), F_ULOCK, 0);
 	return lock_map_.erase(it);
 }
-- 
2.51.0

